src/types/context.rs

```diff
@@ -171,10 +171,13 @@ impl Context {
     pub fn bind(&self, existing: &Type, new: Bound, hint: &'static str) -> Result<(), Error> {
         let existing_root = existing.bound.root();
         let mut lock = self.lock();
-        lock.bind(existing_root, new).map_err(|e| Error::Bind {
-            existing_bound: e.existing,
-            new_bound: e.new,
-            hint,
+        lock.bind(existing_root, new).map_err(|e| {
+            let new_bound = lock.alloc_bound(Arc::as_ptr(&self.slab), e.new);
```

Comment:
7e7756ebcdde18b28ed0a2193380b8d7cc065e5c: `new_bound` needs to be allocated because the bound doesn't exist in the slab yet. `Type` consists of a `BoundRef` that lives inside the slab, so we have to allocate. Is that correct?

Reply:
Yep exactly.